### ISSUE TITLE
Format stabilization tracking issue

### ISSUE THREAD

**Issue by casey** [casey/just#868]
I've landed a few PRs that improve `--fmt` output recently:

- #3275
- #3274
- #3268
- #3282

I now have a bit of a dilemma. I'm starting to think that formatting output is good enough that the feature could be stabilized. However, I think that formatting is a feature which is likely to be subject to tweaking over the long term. Better ways to format the same thing (like breaking long expressions onto multiple lines), and fixing random bugs.

Given all that, there are a few options:

1. `--fmt` is permanently unstable
2. `--fmt` is stable but can change in a minor just release and isn't bound by just's normal stability guarantee (y in `x.y.z`)
3. `--fmt` is stable but we can't ever improve it

I think that 2. is probably the best option, but it's a departure from the usual extremely strong stability guarantees. If we did 2., I would probably still try to avoid frequent changes, which, most annoyingly would break CI jobs that use `--fmt --check`, maybe keeping them to once or twice per year.

Let me know what you think in!

**Comment by casey** [comment]
Would people prefer that formatting stabilize but that it can change in a minor (y in `x.y.z`) version release? I think that if just's backwards compatibility requirements mean that the output of `just --fmt` for a given justfile can never changed, it will be almost possible to ever stabilize `--fmt`, or it will be stable but not ideal, and there will be improvements that we want to make but can't.

**Comment by smorin** [comment]
Agree with Option 2 for what it's worth @casey

**Comment by valscion** [comment]
Yeah option 2 sounds the best but I'd not restrict format bumps only to minors if the formatter output would be broken. Buggy formatting could still go in regular patch versions as long as its blast radius is likely small. Kind of similar versioning then as Prettier does in JS side

**Comment by casey** [comment]
Finally stabilized in [1.50.0](https://github.com/casey/just/releases/tag/1.50.0)! The output of `--fmt` is not bound by Just's backwards compatibility guarantees, but I'll try to not go crazy changing it too often.

**Comment by Sander-Brickmans** [comment]
Would it be possible to add a --fmt-version to the --fmt command so you can keep it backwards-compatible whilst also allowing improvements over time? The lack of the `--fmt-version` flag would use latest.

**Comment by casey** [comment]
I like the idea, but in some cases it would be very hard to preserve both the old and new behavior. For example, the change to the way that newlines are handled between items was pretty involved, and having to keep the old version around would be pretty annoying.
